### DISCUSSION TITLE
Password salts

### DISCUSSION THREAD

**Discussion by florian-klemt** [blitz-js/blitz#4249] [General] [answered]
Hi everyone,

I was wondering if passwords are salted by default. I don't see anything like that, so I presume no, but wanted to check first.

**Comment by MrLeebo** [comment, marked as answer]
Yes, they are salted. https://github.com/blitz-js/blitz/blob/fb232d126ebdbb9183417336a4ce461dc89b2a56/packages/blitz-auth/src/server/secure-password.ts#L12 the `hash()` function of `secure-password` applies the salt.

**Reply by florian-klemt** [reply]
Ok, great 👍🏼
Just out of curiosity, shouldn't the salt be also stored in the db? How does it work in detail?

**Reply by MrLeebo** [reply]
They're both stored sequentially in the same field. Essentially, it is this:

```ts
byteArray = [...salt, ...hash]
```

The db field is a byte array, the first 16 bytes are the salt. The remaining bytes are the password hash.

**Reply by florian-klemt** [reply]
Thank you very much
